src/streams.cpp

```diff
@@ -7,18 +7,24 @@
 
 #include <array>
 
+AutoFile::AutoFile(std::FILE* file, std::vector<std::byte> data_xor)
+    : m_file{file}, m_xor{std::move(data_xor)}, m_position{0}
+{
+    if (!IsNull()) {
+        auto pos{std::ftell(m_file)};
+        if (pos >= 0) m_position = pos;
```

Comment:
nit: Handle ftell error

```suggestion
        if (pos < 0) {
            throw std::ios_base::failure("AutoFile::seek: ftell failed");
        }
        m_position = pos;
```

Reply:
That makes sense, I didn't consider pipes/fifos. 

I agree on this approach given that a bad `m_position` value is only an issue when XOR'ing, and your branch has checks for an unknown `m_position` before any XOR operations. And in general, it seems to me, `ftell`'s are followed by file operations that fail if the `ftell` fails for any reason other than the file being a pipe/fifo (`ESPIPE`).

I think ideally `errno` is checked here for values other than `ESPIPE`, but the current approach is good.